<launch>

    <!-- Start the shelf detection node -->
    <arg name="SOURCE" default="/zed2/zed_node/rgb/image_rect_color" />
    <arg name="CAMERA_INFO" default="/zed2/zed_node/depth/camera_info" />
    <arg name="DEPTH_INPUT" default="/zed2/zed_node/depth/depth_registered" />
    <arg name="CAMERA_FRAME" default="zed2_left_camera_optical_frame" />
    <arg name="MIN_CLUSTERS" default="2" />
    <arg name="MAX_CLUSTERS" default="6" />

    <node pkg="vision" type="ShelfDetection.py" name="shelf_detection" respawn="true" output="screen">
        <param name="SOURCE" value="$(arg SOURCE)" />
        <param name="CAMERA_INFO" value="$(arg CAMERA_INFO)" />
        <param name="DEPTH_INPUT" value="$(arg DEPTH_INPUT)" />
        <param name="CAMERA_FRAME" value="$(arg CAMERA_FRAME)" />
        <param name="MIN_CLUSTERS" value="$(arg MIN_CLUSTERS)" />
        <param name="MAX_CLUSTERS" value="$(arg MAX_CLUSTERS)" />
    </node>

</launch>